src/main/java/es/degrassi/mmreborn/common/manager/ComponentManager.java

```diff
@@ -62,25 +67,37 @@ public final void reset() {
   }
 
   public final void updateModifiers(boolean force) {
-    if (controller.getFoundMachine() == DynamicMachine.DUMMY) return;
-    if (controller.getLevel() == null) return;
-    if (force || controller.getLevel().getGameTime() % MMRConfig.get().checkStructureTicks.get() == 0) {
-      foundModifiers.clear();
-      foundModifiers.putAll(gatherModifiers());
-    }
+    if (controller.getFoundMachine() == DynamicMachine.DUMMY)
+      return;
+    Level level = controller.getLevel();
+    if (level == null)
+      return;
+    long gameTime = level.getGameTime();
+    if (!Utils.shouldRunPeriodicCheck(force, gameTime, lastModifiersCheckTick, tickOffset,
+        MMRConfig.get().checkStructureTicks.get()))
+      return;
+    lastModifiersCheckTick = gameTime;
+    foundModifiers.clear();
+    foundModifiers.putAll(gatherModifiers());
     controller.setChanged();
```

Comment:
There is a subtle change here and in `updateComponents` - before the proposed changes, `controller.setChanged()` would run even if the periodic check wasn't run, while now we don't do that anymore. I couldn't find any issues during testing, but could there be any ?